File: backend/models/task.go
package models

import (
	"time"

	"github.com/jinzhu/gorm"
)

type Task struct {
	gorm.Model
	UserID      uint      `json:"user_id" gorm:"not null"`
	User        User      `json:"-" gorm:"foreignkey:UserID"`
	Title       string    `json:"title" binding:"required"`
	Description string    `json:"description"`
	Status      string    `json:"status" binding:"required,oneof=Pending In-Progress Completed"`
	Priority    string    `json:"priority" binding:"required,oneof=Low Medium High Critical"`
	DueDate     time.Time `json:"due_date"`
}

File: frontend/src/types/task.ts
export interface Task {
  // Support both uppercase (frontend) and lowercase (backend) field names
  // ID fields
  ID?: number;
  id?: number;
  
  // Timestamp fields
  CreatedAt?: string;
  created_at?: string;
  UpdatedAt?: string;
  updated_at?: string;
  DeletedAt?: string | null;
  deleted_at?: string | null;
  
  // Task data fields
  Title?: string;
  title?: string;
  Description?: string;
  description?: string;
  Status?: "Pending" | "In-Progress" | "Completed";
  status?: string;
  Priority?: "Low" | "Medium" | "High" | "Critical";
  priority?: string;
  DueDate?: string;
  due_date?: string;
  user_id?: number;
  
  // Tags field removed as it's causing SQL issues
}
